daktari/checks/intellij_idea.py

```diff
@@ -186,6 +186,33 @@ def validate_query_result(self, result):
         return current_package_manager.__contains__(self.package_manager_path)
 
 
+class IntelliJTypescriptCompilerPathConfigured(XmlFileXPathCheck):
+    name = "intellij.typescriptCompilerPathConfigured"
+    file_path = ".idea/compiler.xml"
+    xpath_query = "./component[@name='TypeScriptCompiler']/option[@name='typeScriptServiceDirectory']"
+    depends_on = [IntelliJProjectImported]
+
+    def __init__(self, typescript_compiler_path: str):
+        self.typescript_compiler_path = typescript_compiler_path
+        resolved_path = typescript_compiler_path.replace("$PROJECT_DIR$", os.getcwd())
+        self.pass_fail_message = f"IntelliJ typescript compiler path has <not/> been set to {resolved_path}"
+
+        self.suggestions = {
+            OS.GENERIC: f"""
+                Follow the steps to set {resolved_path} as your typescript compiler path:
+                https://www.jetbrains.com/help/idea/typescript-support.html#ws_ts_use_ts_service_checkbox
+                """
+        }
+
+    def validate_query_result(self, result):
+        if result is None:
+            self.pass_fail_message = "IntelliJ typescript compiler path is <not/> set"
+            return False
+        current_typescript_compiler_path = result.get('value')
+        logging.debug(f"IntelliJ typescript compiler set to: {current_typescript_compiler_path}")
+        return current_typescript_compiler_path.__eq__(self.typescript_compiler_path)
```

Comment:
The PR bot was actually helpful for once and suggests the equality operator:
```suggestion
        return current_typescript_compiler_path == self.typescript_compiler_path
```